bin/locale.php

```diff
@@ -55,14 +61,155 @@
 $storage = [];
 $view->addExtension(new TwigMessages(new Messages($storage)));
 
-
 foreach (glob(PROJECT_ROOT . '/views/*.twig') as $file) {
     echo var_export($file, true) . PHP_EOL;
 
     $view->getEnvironment()->load(str_replace(PROJECT_ROOT . '/views/', '', $file));
 }
-foreach (glob(PROJECT_ROOT . '/modules/*.twig') as $file) {
-    echo var_export($file, true) . PHP_EOL;
 
-    $view->getEnvironment()->load(str_replace(PROJECT_ROOT . '/modules/', '', $file));
-}
\ No newline at end of file
+/**
+ * Mock PDO Storage Service which returns an empty array when select queried.
+ */
+class MockPdoStorageServiceForModuleFactory extends \Xibo\Storage\PdoStorageService
+{
+    public function select($sql, $params, $connection = 'default', $reconnect = false, $close = false)
+    {
+        return [];
+    }
+}
+
+// Mock Config Service
+class MockConfigService extends ConfigService
+{
+    public function getSetting($setting, $default = null, $full = false)
+    {
+        return '';
+    }
+}
+
+// Transactor funtion
+function __($original)
+{
+    return $original;
+}
+
+// Stash
+$pool = new \Stash\Pool();
+
+// Create a new Sanitizer service
+$sanitizerService = new \Xibo\Helper\SanitizerService();
+
+// Create a new base dependency service
+$baseDepenencyService = new \Xibo\Service\BaseDependenciesService();
+$baseDepenencyService->setConfig(new MockConfigService());
+$baseDepenencyService->setStore(new MockPdoStorageServiceForModuleFactory());
+$baseDepenencyService->setSanitizer($sanitizerService);
+
+$moduleFactory = new \Xibo\Factory\ModuleFactory(
+    '',
+    $pool,
+    $view,
+    new MockConfigService(),
+);
+$moduleFactory->useBaseDependenciesService($baseDepenencyService);
+// Get all module
+$modules = $moduleFactory->getAll();
+
+$moduleTemplateFactory = new \Xibo\Factory\ModuleTemplateFactory(
+    $pool,
+    $view,
+);
+$moduleTemplateFactory->useBaseDependenciesService($baseDepenencyService);
+// Get all module templates
+$moduleTemplates = $moduleTemplateFactory->getAll();
+
+// --------------
+// Create translation file
+// Each line contains title or description or properties of the module/templates
+$file = PROJECT_ROOT. '/locale/moduletranslate.php';
+if (!file_exists($file)) {
```

Comment:
I don't think you need to do this - file_put_contents will create a file if it doesn't exist and overwrite one if it does.